main.js

```diff
@@ -1,41 +1,77 @@
 import youtubeRegex from 'youtube-regex'
 import providersList from './src/providers'
 
-const youtubeUrlToId = (url) => {
+/*
+	Media provider URL parsers
+*/
+
+const parseFileUrl = (url) => {
+	let result = new URL(url)
+	let s = result.pathname.split('/')
+	let id = s[s.length - 1]
```

Comment:
and the id of the file is, from the full url splitted at `/`, the last element of the `s` array